### ISSUE TITLE
ECS CodeDeploy rolling deployments

### ISSUE THREAD

**Issue by clareliguori** [aws/containers-roadmap#288]
Similar to [ECS blue-green deployments with AWS CodeDeploy](https://aws.amazon.com/blogs/devops/use-aws-codedeploy-to-implement-blue-green-deployments-for-aws-fargate-and-amazon-ecs/), but supporting non-load-balanced services such as queue pollers and data processors.  CodeDeploy rolling deployments would monitor alarms for automatic rollbacks, support validation hooks, and support a variety of deployment configurations ([examples here](https://docs.aws.amazon.com/codedeploy/latest/userguide/deployment-configurations.html#deployment-configuration-lambda)).

**Comment by nicwise** [comment]
Yes please. But please don't limit it to things which don't have an ALB :) Thanks!

**Comment by nathanielks** [comment]
Agreed, I'm being bit by this now. It would be great to support this!

**Comment by nakulpathak3** [comment]
Agreed as well. Having rolling deploys in CodeDeploy would also unlock sending deploy notifications based on deploy start/end cloudwatch events. Using services-stable CloudWatch event does not work since that event fires during non-deploys too.

**Comment by anagarjunr** [comment]
Waiting to see this

**Comment by jhg023** [comment]
Any updates on this? I'm looking to replace Jenkins with CodeBuild+CodeDeploy, but this issue is blocking the transition.

**Comment by vibhav-ag** [comment]
I had a related question: what are the most important improvements that you would like to see in the existing native rolling update functionality? A few that I see on this thread are deployment notifications, validation hooks, and alarm-based rollbacks. Are there others? Which ones are most important for your use case?

**Comment by efrodriguez** [comment]
I would like rolling updates to have a deployment history list where previous deployments are selectable in order to initiate rollbacks. It would be similar to the deployment history list that CodeDeploy has for Blue/Green deployments. This would make rollbacks super straightforward.
